src/braket/circuits/circuit_helpers.py

```diff
@@ -23,10 +23,12 @@ def validate_circuit_and_shots(circuit: Circuit, shots: int) -> None:
         shots (int): shots to validate
 
     Raises:
-        ValueError: If no result types specified for circuit and `shots=0`.
-            See `braket.circuit.result_types`. Or, if `StateVector` or `Amplitude`
-            are specified as result types when `shots > 0`.
+        ValueError: If circuit has no instructions. Also, if no result types
+            specified for circuit and `shots=0`. See `braket.circuit.result_types`.
+            Or, if `StateVector` or `Amplitude` are specified as result types when `shots > 0`.
```

Comment:
Is the check for `Sample` with `shots=0` done somewhere else? I forgot. If it is, does it make sense to move it here?

Reply:
There's already a check for that in the default simulator and API. I don't think we need to add a duplicate check here.